processor.go

```diff
@@ -305,79 +314,64 @@ func (p *Processor[AC, OC, JC]) shutdown() {
 	close(p.returnChan)
 }
 
-func (p *Processor[AC, OC, JC]) kickJobs(rtn Return[JC], j Job[JC], r *Run[OC, JC]) {
-	if rtn.KickRequests == nil {
-		return
-	}
-	for _, k := range rtn.KickRequests {
-		// create a new job with the right state
-		newJob := Job[JC]{
-			Id:          fmt.Sprintf("%s->%d", j.Id, len(r.Jobs)),
-			C:           k.C,
-			State:       k.State,
-			StateErrors: map[string][]string{},
-		}
-
-		// validate it
-		_, ok := p.stateMap[newJob.State]
-		if !ok {
-			log.Fatal(p.invalidStateError(newJob.State))
-		}
-
-		// return it to the run, it'll get re-enqueued by the main return loop
-		r.Return(newJob)
-	}
-}
-
 type StateExec[AC any, OC any, JC any] struct {
-	State      State[AC, OC, JC]
-	i          int
-	wg         sync.WaitGroup
-	c          chan Job[JC]
-	Overall    OC
 	ctx        context.Context
-	returnChan chan Return[JC]
 	ac         AC
+	oc         OC
```

Comment:
One of my thoughts was that this struct was probbably a better place to keep track of in flight work for any given state... it knows when it's:
- Got a job
- Waiting for rate limiter
- Processing a job
- Waiting to return the job

Which is I think useful when optimizing the framework but also concurrency.

Also lets the processor not have to know as much details or maintain the central counts for the states